packages/engine/Source/Scene/Model/MetadataPickingPipelineStage.js

```diff
@@ -0,0 +1,84 @@
+import ShaderDestination from "../../Renderer/ShaderDestination.js";
+
+/**
+ * The MetadataPickingPipelineStage is inserting the
+ * metadataPickingStage function into the shader code,
+ * including the 'defines' that will be filled with
+ * the proper values for metadata picking in the
+ * 'DerivedCommands'
+ *
+ * @namespace MetadataPickingPipelineStage
+ * @private
+ */
+const MetadataPickingPipelineStage = {
+  name: "MetadataPickingPipelineStage", // Helps with debugging
+};
+
+/**
+ * Process a primitive. This modifies the following parts of the render resources:
+ * <ul>
+ *  <li>adds the required defines and "metadataPickingStage" to function in the shader</li>
+ * </ul>
+ *
+ * @param {PrimitiveRenderResources} renderResources The render resources for this primitive.
+ * @param {ModelComponents.Primitive} primitive The primitive.
+ * @param {FrameState} frameState The frame state.
+ */
+MetadataPickingPipelineStage.process = function (
+  renderResources,
+  primitive,
+  frameState
+) {
+  const shaderBuilder = renderResources.shaderBuilder;
+
+  shaderBuilder.addDefine(
+    "METADATA_PICKING_VALUE_TYPE",
+    "float",
+    ShaderDestination.FRAGMENT
+  );
+  shaderBuilder.addDefine(
+    "METADATA_PICKING_VALUE_STRING",
+    "0.0",
+    ShaderDestination.FRAGMENT
+  );
+  shaderBuilder.addDefine(
+    "METADATA_PICKING_VALUE_COMPONENT_X",
```

Comment:
That 'somewhere' seems to be the respective pipeline stage. I now changed this to use these "enum values" (constants). But since this requires consistency _across_ JS- and GLSL code, it's a judgement call whether this also should be used for inlined code, as in https://github.com/CesiumGS/cesium/pull/12075/commits/fe4c2bd0c905397b557a9980d9b7c0c276d76a95#diff-3f2de87fcdd54f9004f45955930d9b5cdac4d74db7911c64460d9d44a20d8502R85 (it is a bit less readable, but ... probably still makes sense...)